### ISSUE TITLE
GCP - Cloud launcer Gate service does not start up

### ISSUE THREAD

**Issue by szantopeter** [spinnaker/spinnaker#1315]
I installed Spinnaker using https://cloud.google.com/launcher/solution/click-to-deploy-images/spinnaker?q=spinnaker

I completed all instructions then I opened localhost:9000 . In the browser console I see all request to localhost:8084 failing. Looking at the gate logs it seems like the application wasn't starting up. The log contains only this :

                  .   ____          _            __ _ _
                 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
                ( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
                 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
                  '  |____| .__|_| |_|_| |_\__, | / / / /
                 =========|_|==============|___/=/_/_/_/
                 :: Spring Boot ::        (v1.2.8.RELEASE)
                
                2016-11-22 15:23:06.952  INFO 1394 --- [           main] com.netflix.spinnaker.gate.Main          : Starting Main v2.95.0 on spinnaker-1 with PID 1394 (/opt/gate/lib/gate-web-2.95.0.jar started by spinnaker in /)
                2016-11-22 15:23:07.301  INFO 1394 --- [           main] ationConfigEmbeddedWebApplicationContext : Refreshing  org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@2364305a: startup date [Tue Nov 22 15:23:07 UTC 2016]; root of context hierarchy

**Comment by jtk54** [comment]
@szantopeter I'll run through the c2d deployment and see if I can reproduce.

**Comment by jtk54** [comment]
I was able to launch an instance from the same c2d solution. Can you `tail -f /var/log/spinnaker/gate/gate.log` and verify no more output statements have shown up? Spinnaker takes a bit (30s or so) to start up.

**Comment by szantopeter** [comment]
I was waiting for a few hour, but didn't help. I deleted the spinnaker instance reinstalled it, and have the same result, application's don't start up. Just out of curiosity when I started up manually

    java -jar /opt/gate/lib/gate-web-2.95.0.jar 

I saw more log message in the console (app failed with an exception this way)

I am not sure what may be different in our cases, but I am using a "free tier" account. Here is my full config. 

    type	
    compute.v1.instance
    disks	
    - autoDelete: true
      boot: true
      deviceName: boot
      initializeParams:
        diskSizeGb: 10.0
        diskType: https://www.googleapis.com/compute/v1/projects/rare-drummer-150311/zones/us-central1-f/diskTypes/pd-standard
        sourceImage: https://www.googleapis.com/compute/v1/projects/click-to-deploy-images/global/images/spinnaker-latest-stable-licensed
      type: PERSISTENT
    machineType	
    https://www.googleapis.com/compute/v1/projects/rare-drummer-150311/zones/us-central1-f/machineTypes/n1-highmem-4
    metadata	
    items:
    - key: startup-script
      value: /opt/spinnaker/install/first_google_boot.sh --status_prefix=c2d-StatusUpdate
    - key: managed_project_credentials
      value: ''
    - key: kube_cluster
      value: cluster-1
    - key: kube_zone
      value: us-central1-c
    - key: gcr_location
      value: https://gcr.io
    - key: gcr_enabled
      value: 'true'
    - key: spinnaker_local
      value: "providers:\n  aws:\n    enabled: False\n    defaultIAMRole: BaseIAMRole\n\
        \    primaryCredentials:\n      name: my-aws-account\n\n  google:\n    enabled:\
        \ true\n    primaryCredentials:\n      name: my-google-account\n    \n      project:\
        \ rare-drummer-150311\n    \n\nservices:\n  default:\n    primaryAccountName:\
        \ ${providers.google.primaryCredentials.name}\n  jenkins:\n    enabled: True\n\
        \    defaultMaster:\n    \n      baseUrl: http://104.198.16.160//\n    \n    \
        \  username: user\n      password: REMOVED\n\n  igor:\n    enabled: True\n\n\
        \  rush:\n    primaryAccount: ${services.default.primaryAccountName}\n"
    networkInterfaces	
    - accessConfigs:
      - name: External NAT
        type: ONE_TO_ONE_NAT
      network: https://www.googleapis.com/compute/v1/projects/rare-drummer-150311/global/networks/default
      subnetwork: https://www.googleapis.com/compute/v1/projects/rare-drummer-150311/regions/us-central1/subnetworks/default
    serviceAccounts	
    - email: default
      scopes:
      - https://www.googleapis.com/auth/cloud-platform
    tags	
    items:
    - spinnaker
    - deployed-from-google-developer-console
    zone	
    us-central1-f

**Comment by szantopeter** [comment]
@jtk54 I found the problem and it may be a bug.

After I manually executed 

    /opt/spinnaker/install/first_google_boot.sh --status_prefix=c2d-StatusUpdate

I got the error message 

    "message": "Google Identity and Access Management (IAM) API has not been used in project rare-drummer-150311 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/iam.googleapis.com/overview?project=rare-drummer-150311 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.",

I opened the URL in the error message, enabled IAM, then spinnaker started to work. As I mentioned I am using brand new GCP Account, and it seems this prerequisite is not documented anywhere.

**Comment by jtk54** [comment]
Ok, I can update the codelab docs to reflect that. Thanks for following up.

**Comment by jtk54** [comment]
I think I'm conflating two things. This was launched from the Cloud Launcher Marketplace, right?

It turns out that I didn't have the AMI api enabled either, but my deployment worked fine. From your config it seems like you're targeting Kubernetes only, is that the case?

**Comment by szantopeter** [comment]
Yes, I launched from the cloud marketplace. I tried with kubernetes only and aws + kubernetes too, the result was the same. Maybe in your GCP account something is set differently than in a fresh account. If you create a new GCP account I think you will be able to reproduce it.

Additional and unrelated note. I am using windows + git bash and the gcloud compute ssh command did not work as it was written :

    gcloud compute ssh --ssh-flag=-L9000:localhost:9000  --ssh-flag=-L8084:localhost:8084 --ssh-flag=-L8087:localhost:8087 --project=rare-drummer-150311 --zone=us-central1-f spinnaker-1

Only after I changed the below parts like this :

    --ssh-flag="-L 9000:localhost:9000"

maybe for the sanity of other poor windows users like me, this should be noted as well.

**Comment by jtk54** [comment]
Yes I think there's some additional setup for Cloud Launcher with Kubernetes that's not documented. I'll look into documenting that in the Cloud Launcher page.

Where was the `--ssh-flag` command without the space displayed? That would be quite confusing for someone not used to using that. Seems like an easy docs fix.

**Comment by szantopeter** [comment]
In the cloud launcher when I open the spinnaker instance I get an overview with some instructions. In my case the URL is https://console.cloud.google.com/deployments/details/spinnaker-1?project=rare-drummer-150311 the ssh command was working on Linux, but not on Windows

**Comment by jtk54** [comment]
Ok, the docs have been updated @szantopeter. Can you verify the issues you saw were addressed? Will close this after you verify.

**Comment by szantopeter** [comment]
@jtk54 Where should I see the documentation update? I don't see any change in the cloud launcher page https://console.cloud.google.com/deployments/details/spinnaker-1?project=rare-drummer-150311

I am not unable to verify if the "enable billing" issue because on my account billing is already enabled.

**Comment by jtk54** [comment]
Actually, I misspoke. The change was submitted yesterday, but it takes a week or so for the change to be published. Let's follow back up then.

**Comment by szantopeter** [comment]
ok, thanks!

**Comment by hekike** [comment]
@szantopeter did it work for you at the end of the day?
Seems to be I'm facing the very same issue.

**Comment by skim1420** [comment]
@hekike can you please go into console.cloud.google.com, enable IAM API and try again? Let me know if you need help enabling IAM API.

We're working with the IAM team to enable this API by default; in the meanwhile you'll need to do this manually prior to using the Spinnaker Launcher (aka C2D, Marketplace) offering.

Let me know how it goes.

**Comment by hekike** [comment]
@skim1420 thanks, it solved my issue!

I faced with another issue while Spinnaker updates Kubernetes deployment, I could solve it with manually upgrading Spinnaker to the latest via:

```
sudo apt-get update
sudo apt-get upgrade spinnaker
```

And adding missing `providers.appengine`: https://github.com/spinnaker/spinnaker/blob/master/config/spinnaker.yml#L368
and `services.front50.gcs`: https://github.com/spinnaker/spinnaker/blob/master/config/spinnaker.yml#L118 fields.

**Comment by lwander** [comment]
Looks like this has been resolved.
